Frontend/src/hooks/useUserData.ts

```diff
@@ -0,0 +1,14 @@
+import { useQuery } from '@tanstack/react-query'
+import { getCompetitorsInfo } from '../api/user/post/get_user_info'
+
+export function useUserData(ids: number[]) {
+  return useQuery({
+    queryFn: () => (ids.length > 0 ? getCompetitorsInfo(ids) : []),
+    queryKey: ['user-info', ...ids],
```

Comment:
Super nitpick, but it might be crucial to actually sort the IDs. We have data tables with custom sorting order, so it is a very realistic use-case for our registration code to sometimes call this as `useUserData([3,5,19])` and sometimes as `useUserData([5,3,19])`. Both yield the same result in terms of the underlying data, but currently would not be cached efficiently by Tanstack-Query.